src/Documents.ts

```diff
@@ -776,6 +776,9 @@ export class Class implements Pointer.Library, Pointer.Validator, ObjectSchema.R
 				persistedProtectedDocument,
 				response,
 			];
+		}, ( error ) => {
+			delete document[ "__CarbonSDK_InProgressOfPersisting" ];
+			throw error;
```

Comment:
Even though a throw statement works, we should try to avoid using it inside functions that return a promise. Please use `return Promise.reject( error );` whenever you want to throw an error.